### ISSUE TITLE
Error on 'test-insight.bitpay.com' - unable to get utxos on testnet.

### ISSUE THREAD

**Issue by neilspaul** [bitpay/bitcore#2241]
When I try to `getUnspentUtxos()` using bitcore-explorer on bitcoin testnet, there is an error. It was working a week before, but now it isn't working. I figured that the host https://test-insight.bitpay.com/ is been for a week now!

Here is my code 

`let bitcore = require('bitcore-lib');
const explorers = require('bitcore-explorers');
const insight = new explorers.Insight('testnet');
insight.getUnspentUtxos(address, function(err, utxos) {
  if (err) {
  	console.log(err);
  } else {
  	console.log(utxos);
}`

when I try this, only the entire body of err is logged on my console. The same code was working a week back.

SNAPSHOT OF TESTNET INSIGHT DOWNTIME :
![Screen Shot 2019-06-18 at 5 11 04 PM](https://user-images.githubusercontent.com/17508390/59679337-258fc800-91ec-11e9-8844-93d39677adb3.png)
![Screen Shot 2019-06-18 at 5 11 28 PM](https://user-images.githubusercontent.com/17508390/59679338-258fc800-91ec-11e9-91e7-16514429d806.png)

THANKS IN ADVANCE!

**Comment by 0x452** [comment]
Same problem.

**Comment by neilspaul** [comment]
Anyone found solution for this??????

**Comment by osagga** [comment]
@neilspaul not sure if this would help, but the v8 version of the bitcore api is still working here:
https://api.bitcore.io/api/BTC/testnet/

Not sure if the API is the same, but you can give it a try

**Comment by danielefavi** [comment]
@neilspaul is right. There is a little problem.

The package `bitcore-explorers` (used by @neilspaul) provides an easy way to get the UTXOs. But the `bitcore-explorers` looks quite outdated.
Using the bitcoin TESTNET the package `bitcore-explorers` gives a server error (*Error 525 SSL handshake failes*) but using the MAINNET everything works fine.

I tried to update the URL in the file [insight.js](https://github.com/bitpay/bitcore-explorers/blob/master/lib/insight.js#LC33) of the package `bitcore-explorers` like below:
```
    url = 'https://api.bitcore.io/api/BTC/testnet';
    // url = 'https://api.bitcore.io';
```
But after several attempts I just receive API errors from *api.bitcore.io* server.

Is there a way to get the UTXOs using `bitcore-lib` avoiding making a direct call to the API server (as @osagga suggested)?

**Comment by neilspaul** [comment]
> @neilspaul not sure if this would help, but the v8 version of the bitcore api is still working here:
> https://api.bitcore.io/api/BTC/testnet/
> 
> Not sure if the API is the same, but you can give it a try:
> 
> for example to list all the utxos for a given address as coin objects, using the following API cal:
> https://api.bitcore.io/api/BTC/testnet/address/my48EN4kDnGEpRZMBfiDS65wdfwfgCGZRz/?unspent=true

@osagga can you share the api documentation link for the v8 of the api, so that I can try using it and see if it works the same way.

**Comment by neilspaul** [comment]
> The package `bitcore-explorers` (used by @neilspaul) provides an easy way to get the UTXOs. But the `bitcore-explorers` looks quite outdated.
> Using the bitcoin TESTNET the package `bitcore-explorers` gives a server error (_Error 525 SSL handshake failes_) but using the MAINNET everything works fine.
> 
> I tried to update the URL in the file [insight.js](https://github.com/bitpay/bitcore-explorers/blob/master/lib/insight.js#LC33) of the package `bitcore-explorers` like below:
> 
> ```
>     url = 'https://api.bitcore.io/api/BTC/testnet';
>     // url = 'https://api.bitcore.io';
> ```
> 
> But after several attempts I just receive API errors from _api.bitcore.io_ server.
> 
> Is there a way to get the UTXOs using `bitcore-lib` avoiding making a direct call to the API server (as @osagga suggested)?

If the api calls work directly, maybe we can write a wrapper using the api endpoints into a node module, so that we don't have to make the api calls directly. Can that work? thoughts?

**Comment by wtesler** [comment]
Still down as of October 2

**Comment by morenoh149** [comment]
I'm using this but not sure if it is safe to do so https://www.npmjs.com/package/bitcore-explorers-up

**Comment by osagga** [comment]
> > @neilspaul not sure if this would help, but the v8 version of the bitcore api is still working here:
> > https://api.bitcore.io/api/BTC/testnet/
> > Not sure if the API is the same, but you can give it a try:
> > for example to list all the utxos for a given address as coin objects, using the following API cal:
> > https://api.bitcore.io/api/BTC/testnet/address/my48EN4kDnGEpRZMBfiDS65wdfwfgCGZRz/?unspent=true
> 
> @osagga can you share the api documentation link for the v8 of the api, so that I can try using it and see if it works the same way.

Yeah the API docs are on this repo here: [packages/bitcore-node/docs/api-documentation.md](/bitpay/bitcore/tree/master/packages/bitcore-node/docs/api-documentation.md)

**Comment by Komal-bansal** [comment]
> @neilspaul not sure if this would help, but the v8 version of the bitcore api is still working here:
> https://api.bitcore.io/api/BTC/testnet/
> 
> Not sure if the API is the same, but you can give it a try:
> 
> for example to list all the utxos for a given address as coin objects, using the following API cal:
> https://api.bitcore.io/api/BTC/testnet/address/my48EN4kDnGEpRZMBfiDS65wdfwfgCGZRz/?unspent=true

I am getting this error when i am passing the utxos fetched by the above api
"Invalid TXID in object"

My code is as follows- 
let bitcore = require("bitcore-lib");

 let bitcore_transaction = new bitcore.Transaction()
        .from(utxos)
        .to(to, amnt)
        .fee(gas)
        .change(from)
        .sign(pk);

**Comment by sarmatdev** [comment]
Issue is still relevant. Does anyone know a solution to the problem?

**Comment by morenoh149** [comment]
I think the explorer apis are no longer being run. look into https://bcoin.io it's still maintained @sarmatdev

**Comment by sarmatdev** [comment]
> I think the explorer apis are no longer being run. look into https://bcoin.io it's still maintained @sarmatdev 

Thanks for the tip!  This is definitely a worthy replacement.
